walk-in-interview/src/main/java/com/google/account/FirebaseAuthSessionCookieUtil.java

```diff
@@ -0,0 +1,137 @@
+    private static final String PROJECT_ID = "com-walk-in-interview";
+
+    public FirebaseAuthSessionCookieUtil() throws IOException {
+        initAdminSDK();
+    }
+
+    /**
+     * Creates the session cookies. It is evoked when the user signs in.
+     *
+     * @param request Log in request.
+     * @return Response with session cookie.
+     */
+    @POST
+    @Path("/business-log-in")
+    @Consumes("application/json")
+    public Response createSessionCookie(HttpServletRequest request) {
+        try {
+            // Gets the ID token sent by the client
+            String idToken = parseIDToken(request);
+
+            // Sets session expiration to 5 days
+            long expiresIn = TimeUnit.DAYS.toMillis(/* duration= */SESSION_COOKIE_DURATION_DAYS);
+            SessionCookieOptions options = SessionCookieOptions.builder().setExpiresIn(expiresIn).build();
+
+            // Creates the session cookie. This will also verify the ID token in the process.
+            // The session cookie will have the same claims as the ID token.
+            String sessionCookie = FirebaseAuth.getInstance().createSessionCookieAsync(idToken, options).get();
+
+            // Sets cookie policy parameters as required
+            NewCookie cookie = new NewCookie(/* name= */ SESSION_COOKIE_NAME, sessionCookie);
+            return Response.ok().cookie(cookie).build();
+        } catch (Exception e) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Failed to create a session cookie").build();
+        }
+    }
+
+    /**
+     * Verifies session cookies.
+     *
+     * @param cookie Session cookie.
+     * @return Response with uid of the current user account.
+     */
+    @POST
+    @Path("/profile")
+    public Response verifySessionCookie(@CookieParam(SESSION_COOKIE_NAME) Cookie cookie) {
+        String sessionCookie = cookie.getValue();
+        try {
+            // Verify the session cookie. In this case an additional check is added to detect
+            // if the user's Firebase session was revoked, user deleted/disabled, etc.
+            FirebaseToken decodedToken = FirebaseAuth.getInstance()
+                    .verifySessionCookie(sessionCookie, /* checkRevoked= */ true);
+            return getUid(decodedToken);
+        } catch (FirebaseAuthException e) {
+            // Session cookie is unavailable, invalid or revoked. Force user to login.
+            return Response.temporaryRedirect(URI.create(LOG_IN_PAGE_PATH)).build();
+        }
+    }
+
+    /**
+     * Clears the session cookie when the user signs out at the client side.
+     *
+     * @param cookie Session cookie
+     * @return Response to direct to log in page.
+     */
+    @POST
+    @Path("/business-log-out")
+    public Response clearSessionCookie(@CookieParam(SESSION_COOKIE_NAME) Cookie cookie) {
+        // The maximum age of the cookie in seconds when the cookie will expire
+        final int maxAge = 0;
```

Comment:
you need to actually clear the firebase session here:
- extract the sessionToken
- clear it with firebase
- return an expired cookie